Packet++/header/WireGuardLayer.h

```diff
@@ -0,0 +1,253 @@
+#pragma once
+
+#include "Layer.h"
+#include "IpAddress.h"
+#include "MacAddress.h"
+
+/// @file
+
+/**
+ * \namespace pcpp
+ * \brief The main namespace for the PcapPlusPlus lib
+ */
+namespace pcpp
+{
+
+	/**
+	 * WireGuard message types
+	 */
+	enum class WireGuardMessageType
+	{
+		HandshakeInitiation = 1,  ///< Handshake Initiation message
+		HandshakeResponse = 2,    ///< Handshake Response message
+		CookieReply = 3,          ///< Cookie Reply message
+		TransportData = 4         ///< Transport Data message
```

Comment:
very nit: some comments are inline (like here), and some are on top of the attribute (like in `wg_common_header`). Maybe move all to be like `wg_common_header` which is what we usually do?

Reply:
Thanks for the good suggestion. I will move all the comments above the attributes.